src/test/java/htsjdk/samtools/SamReaderTest.java

```diff
@@ -169,4 +177,89 @@ public void samRecordFactoryNullHeaderTest() {
         Assert.assertTrue(samRec.getHeader() == null);
     }
 
+    @Test
+    public void testAssertingIteratorUsesLenientOrdering() {
+        //the coordinate comparator's strict sort sorts lower mapping qualities first
+        //so this list is not sorted with respect to that comparator, but it is sorted with respect to the more lenient
+        // file order comparator which only checks the position
+        final List<SAMRecord> looselySorted = Arrays.asList(createRecord(1, 10),
+                                                            createRecord(1, 1),
+                                                            createRecord(2, 1));
+
+        final SAMRecordCoordinateComparator coordinateComparator = new SAMRecordCoordinateComparator();
+
+        //sanity check that this really sorts differently with the file order comparator vs the full ordering coordinate order comparator
+        final List<SAMRecord> sortedWithFileOrderComparator = looselySorted.stream()
+                .sorted(coordinateComparator::fileOrderCompare)
+                .collect(Collectors.toList());
+        Assert.assertEquals(
+                sortedWithFileOrderComparator, looselySorted);
```

Comment:
Not sure why this is wrapped. Is there a standard wrap column for htsjdk code?

Reply:
no, my intellij just got overzealous